### PR TITLE
Improve TimeZone handling of Activity and Error Tabs.

### PR DESCRIPTION
**Pull request by lbwexler**
+ Add new support for working with working with LocalDates, Dates, and TimeZones

+ See hoist-core PR that should be merged with this:
https://github.com/xh/hoist-core/pull/164

+ Also changed the default date range to the current day.  **Happy to revert this unrelated change,** but it is something we discussed, and helpful for an app with a lot of activity where by default you would like to focus on what is happening on the current day.

Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: admin/tabs/activity/clienterrors/ClientErrorsModel.js [hunk 1]
```diff
@@ -216,31 +215,29 @@ export class ClientErrorsModel {
...
+        const appDay = LocalDate.currentAppDay(),
+            start = this.startDay,
+            end = this.endDay,
             diff = end.diff(start),
             incr = diff + 1;
 
         let newStart = start[dir](incr),
             newEnd = end[dir](incr);
 
-        if (newEnd > tomorrow) {
```
Comment (febbraiod) [inline admin/tabs/activity/clienterrors/ClientErrorsModel.js:228]: "tomorrow" was being used to work around the timezone issue? Is that right? Now everyone gets the same app days so we don't need the buffer?

Comment (febbraiod) [reply admin/tabs/activity/clienterrors/ClientErrorsModel.js:228]: NVM I see the comment below.

File: utils/datetime/LocalDate.js [hunk 2]
```diff
@@ -78,6 +80,20 @@ export class LocalDate {
         return this.from(moment());
     }
 
+    /** @returns {LocalDate} - a LocalDate representing the current day in the App TimeZone */
+    static currentAppDay() {
+        const localOffset = (new Date()).getTimezoneOffset() * -1 * MINUTES,
```
Comment (febbraiod) [inline utils/datetime/LocalDate.js:85]: Why multiply by -1 here just to subtract on line 87?  2 -- 1 == 2 + 1

File: admin/tabs/activity/clienterrors/ClientErrorsModel.js [hunk 3]
```diff
@@ -87,26 +88,24 @@ export class ClientErrorsModel {
...
                 {field: 'id', headerName: 'Entry ID', width: 100, align: 'right', hidden: true},
                 {field: 'username', ...usernameCol},
                 {field: 'browser', width: 100},
                 {field: 'device', width: 100},
                 {field: 'userAgent', width: 130, hidden: true},
                 {field: 'appVersion', width: 130},
-                {field: 'appEnvironment', headerName: 'Environment', width: 130},
-                {field: 'msg', headerName: 'User Message', width: 130, hidden: true},
+                {field: 'appEnvironment',  width: 130},
+                {field: 'msg', width: 130, hidden: true},
```
Comment (amcclain) [inline admin/tabs/activity/clienterrors/ClientErrorsModel.js:100]: Thanks for cleaning these up!

**Review by amcclain — APPROVED** [review]

**Pull request merged**